Topic_3/shoppingCart/src/main/java/com/mycompany/app/shoppingCart/ProductDataAccesImp.java

```diff
@@ -0,0 +1,19 @@
+package com.mycompany.app.shoppingCart;
+
+import java.util.ArrayList;
+
+public class ProductDataAccesImp implements ProductDataAcces {
+
+	public ArrayList<Product> getAllProducts() {
+		return DataBaseStub.getListOfProducts();
+	}
+
+	public Product getProduct(int id) {
+		for(Product product : DataBaseStub.getListOfProducts()) {
+			if(product.getId() == id) {
+				return product;
+			}
+		}
+		return null;

```

Comment:
Because it produce NullPointerException. You didn't see this because you are develop whole app.